app/services/support_interface/ministerial_report_export.rb

```diff
@@ -0,0 +1,52 @@
+module SupportInterface
+  class MinisterialReportExport
+    def call
+      results = subject_mapping(subject_status_count)
+
+      rows = {}
+
+      Subject::SUBJECTS.each do |subject|
+        rows[subject] = {
+          applications: 0,
+          offer_received: 0,
+          accepted: 0,
+          application_declined: 0,
+          application_rejected: 0,
+          application_withdrawn: 0,
+        }
+      end
+
+      results.each do |key, count|
+        subject, status = key
+
+        mapped_statuses = Subject::STATUS_MAPPING[status]
+
+        mapped_statuses.each do |mapped_status|
+          rows[:stem][mapped_status] += count if Subject::STEM_SUBJECTS.include? subject
+          rows[:ebacc][mapped_status] += count if Subject::EBACC_SUBJECTS.include? subject
+          rows[:secondary][mapped_status] += count if Subject::SECONDARY_SUBJECTS.include? subject
+          rows[subject][mapped_status] += count
```

Comment:
I think it's worth seeing how performant this is with say 10,000 records. I have a feeling it will be quite slow